src/io/vss_store.rs

```diff
@@ -94,46 +112,122 @@ impl KVStoreSync for VssStore {
 	fn read(
 		&self, primary_namespace: &str, secondary_namespace: &str, key: &str,
 	) -> io::Result<Vec<u8>> {
-		let fut = self.inner.read_internal(primary_namespace, secondary_namespace, key);
-		self.runtime.block_on(fut)
+		let internal_runtime = self.internal_runtime.as_ref().ok_or_else(|| {
+			debug_assert!(false, "Failed to access internal runtime");
+			let msg = format!("Failed to access internal runtime");
+			Error::new(ErrorKind::Other, msg)
+		})?;
+		let primary_namespace = primary_namespace.to_string();
+		let secondary_namespace = secondary_namespace.to_string();
+		let key = key.to_string();
+		let inner = Arc::clone(&self.inner);
+		let fut =
+			async move { inner.read_internal(primary_namespace, secondary_namespace, key).await };
+		// TODO: We could drop the timeout here once we ensured vss-client's Retry logic always
+		// times out.
+		let spawned_fut = internal_runtime.spawn(async move {
+			tokio::time::timeout(VSS_IO_TIMEOUT, fut).await.map_err(|_| {
+				let msg = "VssStore::read timed out";
+				Error::new(ErrorKind::Other, msg)
+			})
+		});
+		self.runtime.block_on(spawned_fut).expect("We should always finish")?
```

Comment:
This is a sync method, so do we really need to worry about an outer runtime that we must call block_on on?

Reply:
Discussed offline. It should be possible to only use the internal runtime here, but in practice this created vague issues around timers, therefore the more complicated solution with two runtimes. It is temporary until the move to full async.